### PR TITLE
Allow configurating history cache non-user context lock timeout

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
- Allow configurating history cache non-user context lock timeout

**Why?**
- Background task used to wait 1s on workflow lock. In 1.21 release we changed this to 500ms and effectively increased the penalty on workflows that schedules multiple activities or childworkflows at once.
- 500ms should be enough but just in case this change causes any negative effect in production, make it configurable so that we don't need to rollback. 
- Since most likely we don't need to change this value, to prevent accessing dynamic config too much in hot code path, changes to this config requires a service restart (actually shard load) to take effect.

**How did you test it?**
- Existing tests.

**Potential risks**
- N/A

**Is hotfix candidate?**
- yes

### CODE REVIEW

File: service/history/workflow/cache/cache.go [hunk 1]
```diff
@@ -220,7 +222,7 @@ func (c *CacheImpl) lockWorkflowExecution(ctx context.Context,
 	if deadline, ok := ctx.Deadline(); ok {
 		var cancel context.CancelFunc
 		if headers.GetCallerInfo(ctx).CallerType != headers.CallerTypeAPI {
```
Comment (yycptt) [inline service/history/workflow/cache/cache.go:224]: @pdoerner Can you send out a separate PR for your operator callerType change to exclude operator from here as well?

**Review by samanbarghi — APPROVED** [review]
LGTM, pending resolving the conflict.
